strax/context.py

```diff
@@ -1462,6 +1462,18 @@ def _get_source_sf(self, run_id, target, should_exist=False):
             raise ValueError('This cannot happen, we just checked that this '
                              'run should be stored?!?')
 
+    @property
+    def provided_dtypes(self):
+        """
+        Summarize useful dtype information provided by this context
+        :return: dictionary of provided dtypes with their corresponding lineage hash and save_when
+        """
+        hashes = set([(d, self.key_for('0', d).lineage_hash, p.save_when)
```

Comment:
While this does work, you might get a different hash depending on the run you might ask for depending on the `default_per_run` argument such as used here:
https://github.com/XENONnT/straxen/blob/master/straxen/plugins/event_processing.py#L293

This kind of way of tracking options is kind of outdated with CMT and also not very nice (exactly for bookkeeping things). However, since it's not forbidden a priori in strax, perhaps you want to make `run_id='0'` a keyword argument?

In case you wonder, for nT we only actively use it for the veto processing (would be better if we can use CMT instead):
`pprint(st.key_for('0', 'records_nv'), st.key_for('13000', 'records_nv'))`

Reply:
Thanks @jorana - I had gotten this code from Jelle at some point a while ago and tbh never realized that the '0' was referring to the runid. I am quite surprised and concerned that the hash for the same context/dtype pair could change depending on the runid. This makes bookkeeping even more of a nightmare than it already is 😕

My personal preference would be to move to CMT for nveto processing ASAP and push to have the collection of dtypes and their hashes be a property of just the context, not the runid. But if this causes problems then yeah we should drop the property decorator and add a kwarg.

Reply:
Yes, for xenon we definitely need to do this, just pointed out that this doesn't have to be the case for strax - a priori (although I don't really like this support, to be honest).

Reply:
Ah yes I see your point. In that case I agree we should probably make this a normal method and add the runid as kwarg. I love properties but oh well I'll live 😁